decred_wallet/Features/Wallet Setup/RecoverExistingWalletViewController.swift

```diff
@@ -111,55 +126,50 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
         if self.validateSeed().valid {
             self.activateConfirmButton()
         } else {
-            self.deactivateConfirmButton()
+            self.deactivateConfirmButtonForError()
         }
     }
     
     func focusSeedWordInput(at tableRowIndex: Int) {
         let tableIndexPath = IndexPath(row: tableRowIndex, section: 0)
-        
         let nextSeedWordCell = self.tableView.cellForRow(at: tableIndexPath) as? RecoveryWalletSeedWordCell
         nextSeedWordCell?.seedWordAutoComplete.becomeFirstResponder()
         
-        self.tableView.scrollToRow(at: tableIndexPath, at: .middle, animated: true)
+        self.tableView.scrollToRow(at: tableIndexPath, at: .bottom, animated: true)
     }
     
     func activateConfirmButton() {
         self.btnConfirm.backgroundColor = UIColor.appColors.decredGreen
-        self.lblEnterAllSeeds.isHidden = true
-        
-        // increase top spacing since warning label is now hidden so as to position button in center
-        self.tableViewFooterTopSpacingConstraint.constant = 30
-        UIView.animate(withDuration: 0.5) {
-            self.tableViewFooter.layoutIfNeeded()
-        }
     }
     
-    func deactivateConfirmButton() {
-        self.btnConfirm.backgroundColor = UIColor.appColors.lightGray
-        self.lblEnterAllSeeds.isHidden = false
-        
-        if self.userEnteredSeedWords.contains("") {
-            self.lblEnterAllSeeds.text = "Not all seeds are entered. Please, check input fields and enter all seeds."
-        } else {
-            self.lblEnterAllSeeds.text = "You entered an incorrect seed. Please check your words."
-        }
-        
-        // reduce top spacing so that warning label and confirm button are centered in display
-        self.tableViewFooterTopSpacingConstraint.constant = 10
-        UIView.animate(withDuration: 0.5) {
-            self.tableViewFooter.layoutIfNeeded()
-        }
+    func deactivateConfirmButtonForError() {
+         self.btnConfirm.backgroundColor = UIColor.appColors.lightGray
     }
     
     @IBAction func backButtonTap(_ sender: Any) {
         self.navigationController?.popViewController(animated: true)
     }
     
     @IBAction func onConfirm() {
-        let validatedSeed = self.validateSeed()
-        if validatedSeed.valid {
-            self.secureWallet(validatedSeed.seed)
+        self.lblEnterAllSeeds.isHidden = false
+        
+        // reduce top spacing so that warning label and confirm button are centered in display
+        self.tableViewFooterTopSpacingConstraint.constant = 10
```

Comment:
This should not always be done when the user taps confirm button, but only if there's an error to display. This can go into a function `displaySeedError(_ errorMessage: String)`. Set the error label text there and do this animation. Of course, you'd only call that function if there's an error to display.

Reply:
ok